File: Resources/lessons/Chapter 4/Examples/HiLow2.java
import java.util.Random;
import java.util.Scanner;

public class HiLow2 {
	
	public static void main(String[] args) {
		
		boolean playing = true;
		Random rand = new Random();
		Scanner scan = new Scanner(System.in);
		System.out.println("Hello and welcome to the High-low number guessing game.");
		
		while (playing) {
			playing = playLoop(rand, scan);
		}
	}
	
	public static boolean playLoop(Random rand, Scanner scan) {
		boolean playing = true;
		int answer = rand.nextInt(100) + 1;
		boolean win = false;
		int numGuesses = 1;
		
		do {
			win = guessLoop(answer, scan);
			if (win) {
				System.out.println("Congrats, you got in " + numGuesses + " guesses.");
			}
			else{
				numGuesses++;
			}
		} while (!win);
		System.out.println("Would you like to play again?(y/n)");
		String playAgain = scan.next();
		if (playAgain.equals("n")) {
			scan.close();
			playing = false;
			System.out.println("Goodbye");
		}
		return playing;
	}
	
	public static boolean guessLoop(int answer, Scanner scan) {
		boolean win = false;
		System.out.print("Guess a number between 1 and 100:");
		int guess = scan.nextInt();
		
		if (guess == answer) {
			win = true;
		}
		else if (guess < answer) {
			System.out.println("Too low");
		}
		else if (guess > answer) {
			System.out.println("Too high");
		}
		return win;
	}
}

File: Resources/lessons/Chapter 4/Examples/HiLow2.py
import random

def playLoop():
	playing = True
	answer = random.randint(1, 100)
	win = False
	numGuesses = 1
	while (win == False):
		win = guessLoop(answer)
		if (win):
			print("Congrats, you got it in", numGuesses, "guesses.")
		else:
			numGuesses += 1
	playAgain = input("Would you like to play again?(y/n):")
	if (playAgain == "n"):
		playing = False
		print("Goodbye")
	return playing

def guessLoop(answer):
	win = False
	guess = int(input("Guess a number between 1 and 100:"))
	if (guess == answer):
		win = True
	elif (guess < answer):
		print("Too low")
	elif (guess > answer):
		print("Too high")
	return win
	
playing = True
print("Hello and welcome to the High-low number guessing game.")
while (playing):
	playing = playLoop()